### ISSUE TITLE
Pictures not displayed if renamed by an external app (and deleted)

### ISSUE THREAD

**Issue by kTyn9A9Kw3** [SimpleMobileTools/Simple-Gallery#18]
Example: On /sdcard/Pictures/ there are 1.jpg, 2.jpg and 3.jpg if 3.jpg is renamed to 4.jpg with a file manager it stops displaying on Simple Gallery even if changed back to 3.jpg or other filename. Now another bug, on Simple Gallery if you rename 2.jpg to 4.jpg (the file we renamed before, and is hidden on Simple Gallery but exists) what happens is 4.jpg becomes 2.jpg and the old file is deleted/replaced. The refreshing feature doesn't help.

**Comment by tibbi** [comment]
are the images displayed correctly in other galleries? It might be a file managers fault too, that it doesnt really refresh media info.

**Comment by kTyn9A9Kw3** [comment]
Good point, it was the file manager causing this. But I tried a couple of file managers and no one refreshes media info when renaming. What I did on another gallery (A Photo Manager) was a media scanning and the renamed (and hidden) picture showed up. Simple Gallery's refresh feature could do this too, or just check the directories for changes.

**Comment by tibbi** [comment]
ye I will probably implement something like that to scan everything, as editing and updating media files is a lot more annoying than it should be

**Comment by tibbi** [comment]
a rescan has been added on request, by Pull to refresh (that pulling from top to down thing)

**Comment by tibbi** [comment]
can you recheck this one with the latest file manager and the gallery? Should be better.

**Comment by kTyn9A9Kw3** [comment]
Sorry for the delay... I did the same test but pull to refresh still doesn't work, only after a reboot the renamed pictures appear on Simple Gallery.

**Comment by kTyn9A9Kw3** [comment]
This was fixed in version 2.3.0 until the recent 2.10.x version, I just had to rename and the gallery was refreshed automatically (no need to pull the thing). I only noticed the fix last month and I was going to close, but now it's broken again :-P

**Comment by tibbi** [comment]
well, I will have to try reproducing it. Im relying on the MediaStore at retrieving images. So if SnapSeed doesnt properly let Android know that a new image has been created, I wont see it. Can you please try adding 'foto' in the included folders at Settings -> Manage Included Folders? Im going through those folders file by file and checking for images, not just relying on the mediastore. Thanks

**Comment by loesvs** [comment]
Hi Tibbi. Thanks for looking into it! 

Apparently, Android and Gallery have no issues knowing a new image has been created: directly after finishing editing, the new file (edited picture) shows up in the SnapSeed folder in the Gallery.

The images only disappear after syncing with FolderSync. It is supposed to copy the file to 'foto' and then delete the original. Whereas Amaze confirms this has indeed happened and I can open the pictures, Gallery apparently can't handle this move. It shows the empty SnapSeed folder with a counter for whatever how many pictures I synced, and the pictures don't show up, even though I included 'foto' in the Included Folders.

How come the pictures _do_ show up after a reboot?

**Comment by loesvs** [comment]
Found it! My sync-app has an option to 'rescan media library' after syncing. If I turn it on, I don't need a reboot. The SnapSeed folder still behaves weirdly (empty but shows up in Gallery with a counter) but I can just exclude that folder.

Thanks for your help, wouldn't have found it without you mentioning the MediaStore.

**Comment by tibbi** [comment]
Alright, that sounds good enough for now :D It might improve in the future, there are still some mysteries around media recognition to me.

**Comment by tibbi** [comment]
should be working already
